sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceTransferJob.cs

```diff
@@ -208,7 +208,7 @@ private async IAsyncEnumerable<JobPartInternal> GetStorageResourcesAsync()
                                 job: this,
                                 partNumber: partNumber,
                                 sourceResource: (StorageResourceItem)current,
-                                destinationResource: _destinationResourceContainer.GetStorageResourceReference(sourceName))
+                                destinationResource: _destinationResourceContainer.GetStorageResourceReference(sourceName, default))
```

Comment:
Should we still pass `sourceResource.resourceId` here?
(Applies to this code in all job types)

Reply:
Done